### ISSUE TITLE
goto_file_vsplit url behavior is unobvious

### ISSUE THREAD

**Issue by l4l** [helix-editor/helix#10621]
For all cases steps are:

1. % # select_all
2. Alt-s # split_selection_on_newline
3. Ctrl-W F # goto_file_vsplit

The strange behavior because once URL is found it stops further processing:

https://github.com/helix-editor/helix/blob/e2594b64c0daf15bacdb5396d439f206fcbe11e6/helix-term/src/commands.rs#L1258-L1260

Besides, if the file is a valid URL it could not be opened. Which may be strange yet the behavior differs with the :open command.

### Case 0

Input:

```
a
b
c
```

Result: opens files `a`, `b`, and `c`.

### Case 1

Input:

```
a
http://example.com
```

Result: opens file `a` and the URL once.

### Case 2

Input:

```
http://example.com
a
```

Result: opens the URL once.

### Case 3

Input:

```
http://example.com
http://example1.com
http://example2.com
```

Result: opens the URL once.

**Comment by pascalkuthe** [comment]
> The strange behavior because once URL is found it stops further processing:

yeah that is a good catch we should probably just use `continue` instead of `return` in that case.

> Besides, if the file is a valid URL it could not be opened. Which may be strange yet the behavior differs with the :open command.

I don't think this is actually something that happens in practice. Nobody names their file `https://...` and in those rare cases its ok to yank the selection and pase into `:open`

**Comment by l4l** [comment]
note, `http://xyz.abc` after normalization is actually a `http:` directory with a file `xyz.abc`, which is still weird though.
